### PR TITLE
Fix crash in packet list

### PR DESCRIPTION
**Pull request by Atlante45**
Fixes a crash when switching domains too fast.

### CODE REVIEW

**Comment by Atlante45** [comment]
Test plan:

- Open JS Console (Ctrl+Alt+J):
  Run: `Window.location = "52.53.174.38"; Window.location = "52.53.235.75";`
  Notice crash on current build but not with this PR.

- Make sure you can still connect to domains
  Try with IP, place names, domain names, etc

**Comment by hifi-gustavo** [comment]
Build console output for commit: b665f38c1f868e9eab9ca13d99b10ec2234b2f3a available at: https://gist.github.com/b66afa38c315fdb8ca5a61299cb959c1<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8215.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8215.dmg

File: libraries/networking/src/NodeList.cpp [hunk 1]
```diff
@@ -513,9 +513,16 @@ void NodeList::processDomainServerConnectionTokenPacket(QSharedPointer<ReceivedM
 
 void NodeList::processDomainServerList(QSharedPointer<ReceivedMessage> message) {
     if (_domainHandler.getSockAddr().isNull()) {
+        qWarning() << "IGNORING DomainList packet while not connect to a Domain Server";
```
Comment (birarda) [inline libraries/networking/src/NodeList.cpp]: `while not connected`

File: libraries/networking/src/NodeList.cpp [hunk 2]
```diff
@@ -513,9 +513,16 @@ void NodeList::processDomainServerConnectionTokenPacket(QSharedPointer<ReceivedM
...
         // refuse to process this packet if we aren't currently connected to the DS
         return;
     }
+    auto domainUuid = _domainHandler.getUUID();
```
Comment (birarda) [inline libraries/networking/src/NodeList.cpp]: `domainUUID`

File: libraries/networking/src/NodeList.cpp [hunk 3]
```diff
@@ -513,9 +513,16 @@ void NodeList::processDomainServerConnectionTokenPacket(QSharedPointer<ReceivedM
...
+    auto messageSourceUuid = message->getSourceID();
```
Comment (birarda) [inline libraries/networking/src/NodeList.cpp]: `messageSourceUUID`

**Comment by hifi-gustavo** [comment]
Build console output for commit: f0fcb94a6ae902a074379e31b7c9685b802488bc available at: https://gist.github.com/b66afa38c315fdb8ca5a61299cb959c1<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8215.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8215.dmg

**Pull request merged**